mlrun/feature_store/retrieval/job.py

```diff
@@ -179,8 +180,15 @@ def to_dataframe(self, columns=None, df_module=None, **kwargs):
                         columns.remove(drop_col)
 
         file_format = kwargs.get("format")
+        target_kind = self.run.status.results["target"]["kind"]
         if not file_format:
-            file_format = self.run.status.results["target"]["kind"]
+            file_format = target_kind
+
+        target_class = kind_to_driver.get(target_kind)
+        if target_class and not target_class.support_pandas:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"to_dataframe does not support targets that do not support pandas engine. Target kind: {target_kind}"
```

Comment:
Not a fan of the phrasing, but I see a bigger problem. At this point, the job has already completed and written the data. The user has already waited (potentially many) minutes for the result. The place to raise this error is before we start the remote job.

Reply:
But we want to let the user write to these targets, we are unable only to do `to_dataframe`.
We can discuss over this topic